src/test/agent_test.rs

```diff
@@ -1,6 +1,7 @@
 use crate::test;
-use std::io::{BufRead, BufReader, Read, Write};
-use std::thread;
+use crate::test::testserver::*;
```

Comment:
I typically prefer to enumerate the imports, or it's hard to see where things come from.